index.tsx

```diff
@@ -17,21 +17,24 @@ import { render, unmountComponentAtNode } from 'react-dom'
  */
 export function react2angular<Props>(
   Class: React.ComponentClass<Props> | React.SFC<Props>,
-  bindingNames?: (keyof Props)[]
+  bindingNames: (keyof Props)[] | null = null,
+  injectNames: string[] = []
 ): IComponentOptions {
   const names = bindingNames
     || (Class.propTypes && Object.keys(Class.propTypes))
     || []
 
   return {
     bindings: fromPairs(names.map(_ => [_, '<'])),
-    controller: ['$element', class extends NgComponent<Props> {
-      constructor(private $element: IAugmentedJQuery) {
+    controller: ['$element', ...(injectNames as any), class extends NgComponent<Props> {
+      injectedProps: any[];
+      constructor(private $element: IAugmentedJQuery, ...injectedProps: any[]) {
         super()
+        this.injectedProps = injectedProps;
       }
       render() {
         // TODO: rm any when https://github.com/Microsoft/TypeScript/pull/13288 is merged
-        render(<Class {...(this.props as any)} />, this.$element[0])
+        render(<Class {...(this.props as any)} {...this.injectedProps} />, this.$element[0])
```

Comment:
Consider reversing the order of `{...this.injectedProps}` and `{...(this.props as any)}`, so `props` take priority.

Reply:
Uh, okay. Fixed the order and added some tests!